From 198e7c4258e44e10ee9af0eba7be87acf119d2d8 Mon Sep 17 00:00:00 2001
From: "Yue, VincentX" <vincentx.yue@intel.com>
Date: Tue, 5 Dec 2017 16:16:53 +0800
Subject: [PATCH 1/2] Remove Help & Feedback option in contacts

Due to AOSP contacts don't support the Help & Feedback
So, don't display  Help & Feedback option if device don't support.

Change-Id: I1843c5310e51e4daa3047d6975faf2931b4f7dcc
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-53511
Signed-off-by: Yue, VincentX <vincentx.yue@intel.com>
---
 src/com/android/contacts/drawer/DrawerAdapter.java         | 7 +++++--
 src/com/android/contacts/editor/ContactEditorFragment.java | 4 ++++
 2 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/src/com/android/contacts/drawer/DrawerAdapter.java b/src/com/android/contacts/drawer/DrawerAdapter.java
index 963c125..420904c 100644
--- a/src/com/android/contacts/drawer/DrawerAdapter.java
+++ b/src/com/android/contacts/drawer/DrawerAdapter.java
@@ -34,6 +34,7 @@ import com.android.contacts.model.account.AccountDisplayInfo;
 import com.android.contacts.model.account.AccountDisplayInfoFactory;
 import com.android.contacts.util.SharedPreferenceUtil;
 import com.android.contactsbind.ObjectFactory;
+import com.android.contactsbind.HelpUtils;
 
 import java.util.ArrayList;
 import java.util.List;
@@ -118,8 +119,10 @@ public class DrawerAdapter extends BaseAdapter {
         mMiscItems.add(new DividerItem());
         mMiscItems.add(new MiscItem(R.id.nav_settings, R.string.menu_settings,
                 R.drawable.quantum_ic_settings_vd_theme_24));
-        mMiscItems.add(new MiscItem(R.id.nav_help, R.string.menu_help,
-                R.drawable.quantum_ic_help_vd_theme_24));
+        if (HelpUtils.isHelpAndFeedbackAvailable()) {
+            mMiscItems.add(new MiscItem(R.id.nav_help, R.string.menu_help,
+                    R.drawable.quantum_ic_help_vd_theme_24));
+        }
         rebuildItemsList();
     }
 
diff --git a/src/com/android/contacts/editor/ContactEditorFragment.java b/src/com/android/contacts/editor/ContactEditorFragment.java
index da68ec6..50eb140 100644
--- a/src/com/android/contacts/editor/ContactEditorFragment.java
+++ b/src/com/android/contacts/editor/ContactEditorFragment.java
@@ -729,12 +729,16 @@ public class ContactEditorFragment extends Fragment implements
         final MenuItem splitMenu = menu.findItem(R.id.menu_split);
         final MenuItem joinMenu = menu.findItem(R.id.menu_join);
         final MenuItem deleteMenu = menu.findItem(R.id.menu_delete);
+        final MenuItem helpMenu = menu.findItem(R.id.menu_help);
 
         // TODO: b/30771904, b/31827701, temporarily disable these items until we get them to work
         // on a raw contact level.
         joinMenu.setVisible(false);
         splitMenu.setVisible(false);
         deleteMenu.setVisible(false);
+        if (!HelpUtils.isHelpAndFeedbackAvailable()) {
+            helpMenu.setVisible(false);
+        }
         // Save menu is invisible when there's only one read only contact in the editor.
         saveMenu.setVisible(!isEditingReadOnlyRawContact());
         if (saveMenu.isVisible()) {
-- 
1.9.1

